### ISSUE TITLE
TreePVector is slow

### ISSUE THREAD

**Issue by blackdrag** [hrldcpr/pcollections#4]
_From [hrld...@gmail.com](https://code.google.com/u/100694777776169671086/) on 2009-03-30T02:18:33Z_

according to tests.Benchmarks, TreePVector is the only PCollection which consistently 
underperforms compared to its corresponding Java API class, ArrayList.

2-3 finger trees might provide a faster implementation, but there might also be some simple 
bottleneck in the current code.

_Original issue: http://code.google.com/p/pcollections/issues/detail?id=4_

**Comment by Groostav** [comment]
According to [this video i found somewhere on the internet](http://www.infoq.com/presentations/Functional-Data-Structures-in-Scala),(and when has a man in a google wave t-shirt ever lead you wrong?) 2-3 finger trees are bad and the better datastructure is the Bitmapped Vector Trie with a large branching factor (32 in clojure) is the way to go.

I would _really_ like to see this.

**Comment by blackdrag** [comment]
yeah, I have that on my hard disc for 6 months already... As soon as I have some free time I will add this
